storage/tianmu/core/value.h

```diff
@@ -18,42 +18,87 @@
 #define TIANMU_CORE_VALUE_H_
 #pragma once
 
+#include <boost/variant.hpp>
 #include <string>
 #include <variant>
 
 namespace Tianmu {
 namespace core {
+// std::variant has some bug in gcc8
```

Comment:
Pls, gives more detail info on this bug.  what behavior will happen on GCC 8.

Reply:
the bug is similar to this but not the same：https://gcc.gnu.org/bugzilla/show_bug.cgi?id=85183 ，core::value has a member which type is  std::variant;  if AttributeTypeInfo has a memeber whose type is core:value, will report many compiler error that can not find matched (copy assignemend?  not the std::variant class ,its internal class), if upgrade to GCC9.3, it work well. GCC8 implementation for std::variant is not stable, So replace  std::variant instead of boost::variant; Keep std::variant for GCC9.3 anyway.